net/FlatBuffers/Table.cs

```diff
@@ -83,15 +84,15 @@ protected int __vector(int offset)
         }
 
         // Initialize any Table-derived type to point to the union at the given offset.
-        protected TTable __union<TTable>(TTable t, int offset) where TTable : Table
+        public T __union<T>(int offset) where T : struct, Accessor
```

Comment:
Release build:

Before change:

    Name                                  Mean   StdD  Unit
    Encode                               2.104  0.872 us/op
    Decode                               0.048  0.071 us/op
    Traverse                             1.008  0.172 us/op

With change:

    Name                                  Mean   StdD  Unit
    Encode                               1.736  0.448 us/op
    Decode                               0.040  0.049 us/op
    Traverse                             1.040  0.165 us/op

Made a bit of a difference.